src/logic/calculate.js

```diff
@@ -0,0 +1,24 @@
+export default function calculate(prevInput, currentInput, operation) {
+  const prevNum = parseFloat(prevInput);
+  const currentNum = parseFloat(currentInput);
+
+  if (Number.isNaN(prevNum) || Number.isNaN(currentNum)) {
+    return 'Error';
+  }
+
+  switch (operation) {
+    case '+':
+      return (prevNum + currentNum).toString();
+    case '-':
+      return (prevNum - currentNum).toString();
+    case 'x':
+      return (prevNum * currentNum).toString();
+    case '÷':
+      if (currentNum === 0) {
+        return 'Error';
+      }
+      return (prevNum / currentNum).toString();
+    default:
+      return 'Error';
```

Comment:
- [ ] You have done greaet so far :+1: :+1: But there are a few requirements you need to meet before moving to the next milestone. :+1: 
- [ ] You have done great adding the `calculator.js` logic file, but it seems you modified the content of the file. As stated in the requirement, you are meant to study and unstand the file and use it as it was provided. you are not meant to change or modify the original file. I'll suggest you get t[he original copy of the file here ](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/calculate.js)and work with it as it is. :+1: :smile: :small_airplane: